src/site/_includes/partials/head.njk

```diff
@@ -36,17 +36,20 @@
 function loadScript(url, type) {
   const s = document.createElement('script');
   s.src = url;
-  if (type) s.type = type;
+  if (type) {
+    s.type = type;
+  }
   if (type === 'module') {
     const pre = document.createElement('link');
     pre.rel = 'modulepreload';
     pre.href = url;
-    document.head.appendChild(pre);
+    document.head.append(pre);
+    {# We use DOMContentLoaded as the loader script is running sync, and inserting a module script here doesn't defer. This brings back normal type="module" behavior. #}
```

Comment:
This looks like it shouldn't work because it's in the middle of JS but I guess it does, the `{#` rule wins.

This is a super minor nit but could this be a JS comment, since we're minifying now and it'll be removed at that point? (It might be useful to disable minification in dev... which actually I might suggest.)